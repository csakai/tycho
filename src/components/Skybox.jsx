import React from 'react';
import TextureContainer from '../containers/TextureContainer';

export default class Skybox extends React.Component {
  render() {
    return (
      <mesh>
        <TextureContainer
          textures={[{
            url: 'milkyway.jpg'
          }]}
        />
        <sphereGeometry
          widthSegments={32}
          heightSegments={32}
          radius={700}
        />
      </mesh>
    );
  }
}
