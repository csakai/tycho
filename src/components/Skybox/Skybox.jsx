import React from 'react';
import TextureContainer from '../../containers/TextureContainer';
import Constants from '../../constants';

export default class Skybox extends React.Component {
  render() {
    return (
      <mesh>
        <TextureContainer
          textures={[{
            url: 'milkyway.jpg'
          }]}
        />
        <sphereGeometry
          widthSegments={Constants.WebGL.SPHERE_SEGMENTS}
          heightSegments={Constants.WebGL.SPHERE_SEGMENTS}
          radius={Constants.WebGL.SKYBOX_RADIUS}
        />
      </mesh>
    );
  }
}
