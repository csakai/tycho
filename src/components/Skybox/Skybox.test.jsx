import React from 'react';
import toJson from 'enzyme-to-json';
import {shallow} from 'enzyme';
import Skybox from './Skybox';

describe('Skybox Component', () => {
  let component;

  beforeEach(() => {
    component = shallow(<Skybox />);
  });

  describe('render()', () => {
    it('should render the Skybox successfully', () => {
      expect(toJson(component)).toMatchSnapshot();
    });
  });
});
