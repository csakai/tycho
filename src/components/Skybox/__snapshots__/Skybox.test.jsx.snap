// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skybox Component render() should render the Skybox successfully 1`] = `
<mesh>
  <TextureContainer
    textures={
      Array [
        Object {
          "url": "milkyway.jpg",
        },
      ]
    }
  />
  <sphereGeometry
    heightSegments={32}
    radius={15000}
    widthSegments={32}
  />
</mesh>
`;
